iotilecore/iotile/core/utilities/async_tools/event_loop.py

```diff
@@ -203,13 +203,13 @@ def add_subtask(self, subtask):
         finished = asyncio.gather(*tasks, return_exceptions=True)
 
         try:
-            await asyncio.wait_for(finished, timeout=self._stop_timeout)
+            outcomes = await asyncio.wait_for(finished, timeout=self._stop_timeout)
         except asyncio.TimeoutError as err:
             # See discussion here: https://github.com/python/asyncio/issues/253#issuecomment-120138132
             # This prevents a nuisance log error message, finished is guaranteed
             # to be cancelled but not awaited when wait_for() has a timeout.
             try:
-                await finished
```

Comment:
Actually I think that you need to explicitly assign `outcomes = []` here.  If there was  a timeout then the `finished` future was canceled automatically by `asyncio.wait_for` so when you await it, it will always raise `asyncio.CancelledError` and `outcomes` will never be set. 

I don't think there's a safe way to get any underlying exceptions from the tasks if there was a timeout.

Reply:
Ah, yea, good catch, i'll make sure I assign it outside of the area of exception so we don't try and reference it should it never get created.